core/codegen/tests/ui-fail/from_param.rs

```diff
@@ -0,0 +1,23 @@
+use rocket::request::FromParam;
+
+#[derive(FromParam)]
+struct Foo1 {
+    a: String
+}
+
+#[derive(FromParam)]
+struct Foo2 {}
+
+#[derive(FromParam)]
+enum Foo3 {
+    A(String),
+    B(String)
+}
+
+#[derive(FromParam)]
+enum Foo4 {}
```

Comment:
Looking at the compiler output, it doesn't look like this derive fails, and I'm not sure whether it should. I think we should consider this a supported case, since it's useful for prototyping, but I'm open to arguments either way.

Reply:
I don't really have any opinion on it so let's allow it.